packages/@sanity/form-builder/src/inputs/InvalidValueInput/InvalidValueInput.tsx

```diff
@@ -73,29 +71,107 @@ export default class InvalidValueInput extends React.PureComponent<InvalidValueP
       )
     }
 
-    const converters = getConverters(value, actualType, validTypes)
-    const message = (
-      <>
-        Encountered a value of type <code>{actualType}</code>.{this.renderValidTypes()}
-        <h4>{actualType}</h4>
-        <pre className={styles.inspectValue}>{JSON.stringify(value, null, 2)}</pre>
-        {converters.map((converter) => (
-          <DefaultButton
-            key={`${converter.from}-${converter.to}`}
-            onClick={() => this.handleConvertTo(converter.convert(value))}
-            color="primary"
-          >
-            Convert value to {converter.to}
-          </DefaultButton>
-        ))}
-        <div className={styles.buttonWrapper}>
-          <DefaultButton onClick={this.handleClearClick} color="danger">
-            Remove value
-          </DefaultButton>
-        </div>
-      </>
+    const suffix = (
+      <Stack padding={2}>
+        <Button onClick={handleClearClick} tone="critical" text="Reset value" />
+      </Stack>
     )
 
-    return <Warning heading="Content has invalid type" message={message} />
+    return (
+      <Alert status="error" suffix={suffix} title={<>Invalid property value</>}>
+        <Text as="p" muted size={1}>
+          The property value is stored as a value type that does not match the expected type.
+        </Text>
+
+        <Details marginTop={4} open={__DEV__} title={<>Developer info</>}>
+          <Stack space={3}>
+            {validTypes.length === 1 && (
+              <Text as="p" muted size={1}>
+                The value of this property must be of type <code>{validTypes[0]}</code> according to
+                the schema.
+              </Text>
+            )}
+
+            {validTypes.length === 1 && (
+              <Text as="p" muted size={1}>
+                Mismatching value types typically occur when the schema has recently been changed.
+              </Text>
+            )}
+
+            {validTypes.length !== 1 && (
+              <Text as="p" muted size={1}>
+                Only the following types are valid here according to schema:
+              </Text>
+            )}
+
+            {validTypes.length !== 1 && (
+              <ul>
```

Comment:
Should this be a `<Stack as="ul">` to have spacing between the text children?

Reply:
Fixed in https://github.com/sanity-io/sanity/commit/3f01b74e1c002950ab13b3e1bfcefc32d51a9417